File: src/main/java/uk/edu/populationfitness/models/genes/FitnessRange.java
package uk.edu.populationfitness.models.genes;

public class FitnessRange {

    // Defines the minimum fitness expected from the fitness function
    public double min_fitness;

    // Defines the max fitness expected from the fitness function
    public double max_fitness;

    public FitnessRange(){
        min_fitness = 0;
        max_fitness = 1;
    }

    public FitnessRange max(double max){
        max_fitness = max;
        return this;
    }

    public FitnessRange min(double min){
        min_fitness = min;
        return this;
    }

    // returns the range as a scale. S(f) = (f - min) / (max - min)
    public double toScale(double fitness){
        return (fitness - min_fitness) / (max_fitness - min_fitness);
    }
}

File: Scala/PopulationFitness/src/main/scala/uk/edu/populationfitness/models/tuning/FitnessRange.scala
package uk.edu.populationfitness.models.tuning

class FitnessRange private[tuning]() {

  // Defines the minimum fitness expected from the fitness function
  private var _min = .0
  // Defines the max fitness expected from the fitness function
  private var _max = 1.0
  private var _range = 1.0

  protected def max_= (max: Double): Unit = {
    _max = max
    _range = _max - _min
  }

  protected def min_= (min: Double): Unit = {
    _min = min
    _range = _max - _min
  }

  def max: Double = _max

  def min: Double = _min
}
